webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueCompilationCache.scala

```diff
@@ -61,56 +84,54 @@ case class CompilationStatusErrors(techniquesInError: NonEmptyChunk[EditorTechni
   */
 trait ReadTechniqueCompilationStatusService {
 
-  def get(): IOResult[CompilationStatus]
+  def get(): IOResult[List[EditorTechniqueResult]]
 
 }
 
 /**
   * Update technique compilation status from technique compilation output and technique info
   */
 trait WriteTechniqueCompilationStatusService {
-
-  def update(technique: EditorTechnique, output: TechniqueCompilationOutput): UIO[CompilationStatus]
-
+  /*
+   * Given new editor technique compilation results, update current status and sync it with UI
+   */
+  def updateAndSync(results: List[EditorTechniqueResult]): IOResult[Unit]
+
+  /*
+   * The whole process that lookup for compilation status and update everything
+   */
+  def getUpdateAndSync(): IOResult[Unit]
 }
 
 /**
   * Service to update global technique compilation status and get the latest one from filesystem
   */
 class TechniqueCompilationStatusService(
-    readTechniques:           EditorTechniqueReader,
-    techniqueCompiler:        TechniqueCompiler,
-    compilationStatusService: WriteTechniqueCompilationStatusService
+    readTechniques:    EditorTechniqueReader,
+    techniqueCompiler: TechniqueCompiler
```

Comment:
the unique responsibility of that service is to gather information and transform into something usable for the repos (error cache). 
It's a service, in the sense that it doesn't have any memory nor state.